homeassistant/components/sensor/history_values.py

```diff
@@ -0,0 +1,369 @@
+            if comp is not None:
+                self._device_class = comp.attributes.get('device_class')
+                if self._unit_of_measurement is None:
+                    self._unit_of_measurement = comp.attributes.get(
+                        CONF_UNIT_OF_MEASUREMENT)
+            await self.async_update()
+        remove_listener = self.hass.bus.async_listen(EVENT_STATE_CHANGED,
+                                                     init_sensor)
+        comp = self.hass.states.get(self._target_entity)
+        if comp is not None:
+            self.hass.bus.async_fire(EVENT_STATE_CHANGED, {
+                ATTR_ENTITY_ID: self._target_entity,
+                ATTR_NEW_STATE: comp,
+            })
+
+        if self._update is not None:
+            _LOGGER.debug("Registering updates: %s", self._update)
+
+            @callback
+            def trigger_update(now):
+                """Trigger an update of the sensor."""
+                self.hass.async_create_task(self.async_update())
+            async_track_time_change(self.hass, trigger_update,
+                                    hour=self._update.get(CONF_HOURS),
+                                    minute=self._update.get(CONF_MINUTES),
+                                    second=self._update.get(CONF_SECONDS))
+
+    async def async_update(self):
+        """Update the sensor."""
+        _LOGGER.debug("Updating %s", self.entity_id)
+        # Get previous values of start and end
+        p_start, p_end = self._period
+
+        # Parse templates
+        await self.update_period()
+        start, end = self._period
+
+        # Convert times to UTC
+        start = dt_util.as_utc(start)
+        end = dt_util.as_utc(end)
+        p_start = dt_util.as_utc(p_start)
+        p_end = dt_util.as_utc(p_end)
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        # Compute integer timestamps
+        start_timestamp = math.floor(dt_util.as_timestamp(start))
+        end_timestamp = math.floor(dt_util.as_timestamp(end))
+        p_start_timestamp = math.floor(dt_util.as_timestamp(p_start))
+        p_end_timestamp = math.floor(dt_util.as_timestamp(p_end))
+        now_timestamp = math.floor(dt_util.as_timestamp(now))
+
+        # If period has not changed and current time after the period end...
+        if (start_timestamp == p_start_timestamp
+                and end_timestamp == p_end_timestamp
+                and end_timestamp <= now_timestamp):
+            # Don't compute anything as the value cannot have changed
+            _LOGGER.debug("Update not necessary.")
+            return
+
+        # Get history between start and end
+        history_states = await self.hass.async_add_job(
+            history.state_changes_during_period, self.hass, start, end,
+            str(self._target_entity))
+        history_list = history_states.get(self._target_entity)
+        if history_list is None:
+            self.value = STATE_UNKNOWN
```

Comment:
don't use STATE_UNKNOWN use None

Reply:
should be in developers.home-assistant.io

Reply:
I haven't been able to find it there...